src/nevermined/Nevermined.ts

```diff
@@ -39,19 +40,30 @@ export class Nevermined extends Instantiable {
     }
     instance.setInstanceConfig(instanceConfig)
 
-    instance.keeper = await Keeper.getInstance(instanceConfig)
-    await instance.keeper.init()
-
     // Nevermined main API
-    instance.accounts = new AccountsApi(instanceConfig)
-    instance.agreements = new AgreementsApi(instanceConfig)
-    instance.assets = new AssetsApi(instanceConfig)
-    instance.compute = new ComputeApi(instanceConfig)
-    instance.nfts1155 = await NFT1155Api.getInstance(instanceConfig, instance.keeper.nftUpgradeable)
-    instance.provenance = new ProvenanceApi(instanceConfig)
-    instance.search = new SearchApi(instanceConfig)
-    instance.services = new ServicesApi(instanceConfig)
-    instance.utils = new UtilsApi(instanceConfig)
+    try {
+      instance.accounts = new AccountsApi(instanceConfig)
+      instance.agreements = new AgreementsApi(instanceConfig)
+      instance.provenance = new ProvenanceApi(instanceConfig)
+      instance.search = new SearchApi(instanceConfig)
+      instance.services = new ServicesApi(instanceConfig)
+      instance.utils = new UtilsApi(instanceConfig)
+      instance.keeper = await Keeper.getInstance(instanceConfig)
+      await instance.keeper.init()
+      instance.assets = new AssetsApi(instanceConfig)
+      instance.compute = new ComputeApi(instanceConfig)
+      instance.nfts1155 = await NFT1155Api.getInstance(
+        instanceConfig,
+        instance.keeper.nftUpgradeable,
+      )
+      instance.isKeeperConnected = true
+    } catch (error) {
+      instance.isKeeperConnected = false
+      Logger.error(error)
+      Logger.error(
+        "Contracts didn't initialize because for the above mentioned reason. Loading SDK in offchain mode...",
+      )
+    }
```

Comment:
Not sure about this block of code because it will stop the initialization with the first error instead of trying to initialize everything. Meaning it depends on the order things are being initialized.

For example if the keeper initialization is done before the searchapi and it fails the search api will not be initialized

Reply:
I put first all theses methods that doesn't need the keeper to work, the others after keeper needs it, yes I can put try catch in each one but I think it is not needed